src/sync.h

```diff
@@ -53,7 +53,7 @@ void LeaveCritical();
 void CheckLastCritical(void* cs, std::string& lockname, const char* guardname, const char* file, int line);
 std::string LocksHeld();
 template <typename MutexType>
-void AssertLockHeldInternal(const char* pszName, const char* pszFile, int nLine, MutexType* cs) ASSERT_EXCLUSIVE_LOCK(cs);
+void AssertLockHeldInternal(const char* pszName, const char* pszFile, int nLine, MutexType* cs);
```

Comment:
in commit 5971934eac13809143c763aa515b1960a7368d74:

why are you removing this? This should come with the appropriate annotation to enforce correct usage and warn about missing compile-time annotations

```suggestion
void AssertLockHeldInternal(const char* pszName, const char* pszFile, int nLine, MutexType* cs) EXCLUSIVE_LOCKS_REQUIRED(cs);
```

Reply:
In fact, adding this annotation and compiling the commit shows one instance of a missing annotation.

Reply:
Ah, it looks like you already did that in the next commit :man_facepalming:  653c8d065d90642ffb2d8d4d49474b4abc67a811

Reply:
Should I reorder/squash commits?

Reply:
No need to reorder, but if you want you can describe the two preparatory commits. Suggested git range-diff for the first two commits:


```
1:  00e537d140 ! 1:  efeaeaec78 Use LockAssertion utility class instead of AssertLockHeld()
    @@ Metadata
      ## Commit message ##
         Use LockAssertion utility class instead of AssertLockHeld()
     
    +    First, add the proper attribute to LockAssertion, according to
    +    https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#assert-capability-and-assert-shared-capability
    +
    +    Then, to prepare for commit "Do not hide compile-time thread safety
    +    warnings", replace AssertLockHeld with LockAssertion where needed.
    +
      ## src/net_processing.cpp ##
     @@ src/net_processing.cpp: static void MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid, CConnman& connma
                  }
    @@ src/net_processing.cpp: void PeerLogicValidation::EvictExtraOutboundPeers(int64_
      
                      // Only disconnect a peer that has been connected to us for
                      // some reasonable fraction of our check-frequency, to give
    +
    + ## src/sync.h ##
    +@@ src/sync.h: public:
    + struct SCOPED_LOCKABLE LockAssertion
    + {
    +     template <typename Mutex>
    +-    explicit LockAssertion(Mutex& mutex) EXCLUSIVE_LOCK_FUNCTION(mutex)
    ++    explicit LockAssertion(Mutex& mutex) ASSERT_EXCLUSIVE_LOCK(mutex)
    +     {
    + #ifdef DEBUG_LOCKORDER
    +         AssertLockHeld(mutex);
2:  0ed94d8f86 ! 2:  f0e821b7ef Add missed thread safety annotations
    @@ Metadata
      ## Commit message ##
         Add missed thread safety annotations
     
    +    This is needed for commit "sync.h: make runtime lock checks require
    +    compile-time lock checks" to pass
    +
      ## src/wallet/scriptpubkeyman.h ##
     @@ src/wallet/scriptpubkeyman.h: private:
          //! keeps track of whether Unlock has run a thorough check before

Reply:
Thanks! Commit messages have been updated.